packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3651,19 +3651,21 @@ export class ContainerRuntime
 	/** Implementation of ISummarizerInternalsProvider.refreshLatestSummaryAck */
 	public async refreshLatestSummaryAck(options: IRefreshSummaryAckOptions) {
 		const { proposalHandle, ackHandle, summaryRefSeq, summaryLogger } = options;
+		// proposalHandle is always passed from RunningSummarizer.
+		assert(proposalHandle !== undefined, "proposalHandle should be available");
 		const readAndParseBlob = async <T>(id: string) => readAndParse<T>(this.storage, id);
-		const result = await this.summarizerNode.refreshLatestSummary(
+		const isSummaryTracked = await this.summarizerNode.refreshLatestSummary(
 			proposalHandle,
 			summaryRefSeq,
 		);
 
 		/**
-		 * When refreshing a summary ack, this check indicates a new ack of a summary that was newer than the
+		 * When refreshing a summary ack, this check indicates a new ack of a summary that is newer than the
 		 * current summary that is tracked, but this summarizer runtime did not produce/track that summary. Thus
 		 * it needs to refresh its state. Today refresh is done by fetching the latest snapshot to update the cache
 		 * and then close as the current main client is likely to be re-elected as the parent summarizer again.
 		 */
-		if (result.latestSummaryUpdated && !result.wasSummaryTracked) {
+		if (!isSummaryTracked && summaryRefSeq > this.summarizerNode.referenceSequenceNumber) {
```

Comment:
nit: Consider inverting this to put the GC call (+early return) under the `if`` and reduce nesting of all this.

Reply:
And it makes me curious if we even need to notify GC in the case where the summary wasn't tracked but was old.  Going to read that code again now.

Reply:
If you pass false the GC code is a no-op.  So this can be

```ts
if (isSummaryTracked) {
	await this.garbageCollector.refreshLatestSummary(isSummaryTracked);
	return;
}

if (summaryRefSeq > this.summarizerNode.referenceSequenceNumber) {
	// the existing fetch-then-close block
}
```

Reply:
side note - no GC test passed false for that param :P